actions/workflows/create_vm_role.yaml

```diff
@@ -1,147 +1,208 @@
+        do:
+          - fail_workflow
+  wait_for_instance:
+    action: aws.ec2_wait_for_state
+    input:
+      instance_id: <% ctx().ec2_instance_id %>
+      state: running
+    next:
+      - when: <% succeeded() %>
+        do:
+          - wait_for_ssh
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  wait_for_ssh:
+    action: linux.wait_for_ssh
+    input:
+      hostname: <% ctx().ec2_instance_private_ip %>
+      username: stanley
+      keyfile: <% ctx().keyfile %>
+      ssh_timeout: 20
+      retries: 30
+    next:
+      - when: <% succeeded() %>
+        do:
+          - add_tags
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  add_tags:
+    action: aws.ec2_create_tags
+    input:
+      resource_ids:
+        - <% ctx().ec2_instance_id %>
+      tags: Name=<% ctx().hostname %>,Creator=<% ctx().creator %>,Role=<% ctx().role %>,Purpose=<% switch(isEmpty(ctx().purpose) => "No Description", ctx().purpose => ctx().purpose) %>
+    next:
+      - when: <% succeeded() %>
+        do:
+          - add_cname
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  add_cname:
+    action: aws.r53_zone_add_cname
+    input:
+      name: <% ctx().hostname %>.<% ctx().dns_zone %>
+      value: <% ctx().ec2_instance_private_dns %>
+      zone: <% ctx().dns_zone %>
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - cname: <% ctx().hostname %>.<% ctx().dns_zone %>
+        do:
+          - get_distro
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  get_distro:
+    action: st2cd.get_distro
+    input:
+      hosts: <% ctx().ec2_instance_private_ip %>
+    next:
+      - when: <% succeeded() %>
+        do:
+          - set_hostname
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  set_hostname:
+    action: st2cd.set_hostname
+    input:
+      hosts: <% ctx().ec2_instance_private_ip %>
+      hostname: <% ctx().cname %>
+      distro: <% task(get_distro).result.get(ctx().ec2_instance_private_ip).stdout %>
+      timeout: 300
+    next:
+      - when: <% succeeded() %>
+        do:
+          - reboot
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  reboot:
+    action: core.remote_sudo
+    input:
+      hosts: <% ctx().ec2_instance_private_ip %>
+      cmd: shutdown -r +1 &
+    next:
+      - when: <% succeeded() %>
+        do:
+          - wait_for_ssh_post_reboot
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  wait_for_ssh_post_reboot:
+    action: linux.wait_for_ssh
+    input:
+      hostname: <% ctx().ec2_instance_private_ip %>
+      username: stanley
+      keyfile: <% ctx().keyfile %>
+      ssh_timeout: 30
+      retries: 10
+    next:
+      - when: <% failed() %>
+        do:
+          - fail_workflow
+  fail_workflow:
+    next:
+      - when: <% ctx().ec2_instance_id != null %>
+        do:
+          - destroy_vm_on_failure
+          - fail
```

Comment:
Remove the fail here.  Let the fail in destroy_vm_on_failure to set the workflow status to failed.